metadata/metadata.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metadata
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+)
+
+// AddAnnotations is a method that copies a map into a ReleasePipelineRun.
```

Comment:
can we say it copies a map into a resource's annotations? It doesn't seem to be specific to a release pipelinerun